buildSrc/build.gradle

```diff
@@ -226,6 +226,18 @@ if (project != rootProject) {
     integTestClass = 'org.elasticsearch.gradle.test.GradleIntegrationTestCase'
   }
 
+  testingConventions {
+    naming.clear()
```

Comment:
Why do we need this `clear()`? I don't like the leniency it provides. If we want to provide the ability to replace completely, then let's have a `set` version (much like many task settings can be set with `=` instead of calling DSL-like methods for adding).

Reply:
We do have a set method, so one can write something like `baseClasses = [...]` to replace what was already there.  The `clear` comes from the Gradle container in which the rules live and it's consistent with how Gradle behaves in general. Granted the `clear` method  is not used often.  The intent here is to remove a rule from the container. The default is rule for `Tests` and `IT`. If a project doesn't have both, the alternative is `naming.remove(naming.getByName('Tests'))`.  The semantics of `clear` are I think slightly better, because it expresses "I only have IT", rather than "I don't have Tests", it's also indicative that all the rules are configured right here and there can be nothing coming from configuration elsewhere.  I don't hover have a strong preference.  

I don't like it in general where we have a plugin configure something and then the project throws it out completely.  We sometimes do this with disabling tasks as well.
There should be a way to ask only for what the project needs.  

I also explain in the comment bellow that I view this as an intermediary step and don't want to keep rules as configurable, it's just a means to get to the point where everything follows conventions.